File: Lib/Telerik/Examples/QuickStart/Scheduler/CustomElements/MyAppointmentElement.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using Telerik.WinControls.UI;

namespace Telerik.Examples.WinControls.Scheduler.CustomElements
{
    public class MyAppointmentElement : AppointmentElement
    {
        LightVisualElement icon;

        public MyAppointmentElement(RadScheduler scheduler, SchedulerView view, IEvent appointment) : base(scheduler, view, appointment)
        {
        }

        protected override Type ThemeEffectiveType
        {
            get
            {
                return typeof(AppointmentElement);
            }
        }

        protected override void InitializeAppointment()
        {
            base.InitializeAppointment();

            this.BackColor = ColorTranslator.FromHtml("#FFFCD9");
            this.GradientStyle = Telerik.WinControls.GradientStyles.Solid;

            this.BorderColor = ColorTranslator.FromHtml("#C4C7B6");          
            this.BorderWidth = 3;
        }

        protected override void CreateChildElements()
        {
            base.CreateChildElements();
            
            this.icon = new LightVisualElement();

            this.icon.ShouldHandleMouseInput = false;
            this.icon.NotifyParentOnMouseInput = true;
            this.icon.Alignment = ContentAlignment.BottomRight;
            this.icon.StretchHorizontally = false;
            this.icon.StretchVertically = false;

            this.Children.Add(icon);
        }

        public override void Synchronize()
        {
            base.Synchronize();
            Appointment apointment = this.Appointment as Appointment;
            TimeSpan timeToAppoinmet = this.Appointment.Start - DateTime.Now;
            if (apointment.Reminder != null)
            {
                if (apointment.MasterEvent != null)
                {
                    icon.Margin = new System.Windows.Forms.Padding(0, 0, 10, -4);
                }
                else
                {
                    icon.Margin = new System.Windows.Forms.Padding(0);
                }
                icon.Image = Image.FromFile(@"..\Resources\reminder-icon.png");
                this.icon.ToolTipText = string.Format("Time to apportionment {0} hours and {1} minutes.", timeToAppoinmet.Hours, timeToAppoinmet.Minutes);
            }
            else
            {
                icon.Image = null;
            }
        }
    }
}

File: Lib/Telerik/Examples/QuickStart/Scheduler/CustomElements/MyAppointmentElement.vb
Imports System.Linq
Imports System.Text
Imports Telerik.WinControls.UI

Namespace Telerik.Examples.WinControls.Scheduler.CustomElements
	Public Class MyAppointmentElement
		Inherits AppointmentElement
		Private icon As LightVisualElement

		Public Sub New(ByVal scheduler As RadScheduler, ByVal view As SchedulerView, ByVal appointment As IEvent)
			MyBase.New(scheduler, view, appointment)
		End Sub

		Protected Overrides ReadOnly Property ThemeEffectiveType() As Type
			Get
				Return GetType(AppointmentElement)
			End Get
		End Property

		Protected Overrides Sub InitializeAppointment()
			MyBase.InitializeAppointment()

			Me.BackColor = ColorTranslator.FromHtml("#FFFCD9")
			Me.GradientStyle = Telerik.WinControls.GradientStyles.Solid

			Me.BorderColor = ColorTranslator.FromHtml("#C4C7B6")
			Me.BorderWidth = 3
		End Sub

		Protected Overrides Sub CreateChildElements()
			MyBase.CreateChildElements()

			Me.icon = New LightVisualElement()

			Me.icon.ShouldHandleMouseInput = False
			Me.icon.NotifyParentOnMouseInput = True
			Me.icon.Alignment = ContentAlignment.BottomRight
			Me.icon.StretchHorizontally = False
			Me.icon.StretchVertically = False

			Me.Children.Add(icon)
		End Sub

		Public Overrides Sub Synchronize()
			MyBase.Synchronize()
			Dim apointment As Appointment = TryCast(Me.Appointment, Appointment)
			Dim timeToAppoinmet As TimeSpan = Me.Appointment.Start - Date.Now
			If apointment.Reminder IsNot Nothing Then
				If apointment.MasterEvent IsNot Nothing Then
					icon.Margin = New Padding(0, 0, 10, -4)
				Else
					icon.Margin = New Padding(0)
				End If
				icon.Image = Image.FromFile("..\Resources\reminder-icon.png")
				Me.icon.ToolTipText = String.Format("Time to apportionment {0} hours and {1} minutes.", timeToAppoinmet.Hours, timeToAppoinmet.Minutes)
			Else
				icon.Image = Nothing
			End If
		End Sub
	End Class
End Namespace